src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicSafeHandle.cs

```diff
@@ -92,6 +92,23 @@ internal sealed class MsQuicContextSafeHandle : MsQuicSafeHandle
     /// </summary>
     private readonly MsQuicSafeHandle? _parent;
 
+#if DEBUG
+    /// <summary>
+    /// Native memory to hold TLS secrets. It needs to live same cycle as the underlying connection.
+    /// </summary>
+    private unsafe QUIC_TLS_SECRETS* _tlsSecrets;
+
+    public unsafe QUIC_TLS_SECRETS* GetSecretsBuffer()
+    {
+        if (_tlsSecrets == null)
```

Comment:
My understanding is that it ever happens only once per connection. Should it throw on a second attempt?

Reply:
I don't think it matters as long as it does not leak memory. I'm personally not big fan of throwing for non fatal reasons. I could change that if tat is team preference.